server/bundler/plugins-loader.js

```diff
@@ -0,0 +1,61 @@
+const existsSync = require( 'fs' ).existsSync;
+const readdirSync = require( 'fs' ).readdirSync;
+const path = require( 'path' );
+const camelCase = require( 'lodash' ).camelCase;
+const upperFirst = require( 'lodash' ).upperFirst;
+
+function getReducers( plugins = {} ) {
+	return plugins
+		.map( ( plugin ) => {
+			if ( existsSync( path.join( 'client', 'plugins', plugin, 'state', 'reducer.js' ) ) ) {
```

Comment:
Using this `if` inside of `map` (and later a `filter` to "clean up") seems like an anti-pattern, the simpler form being to filter first based on the `if`'s condition and then map.